src/main/java/bgu/spl/mics/application/MI6Runner.java

```diff
@@ -1,11 +1,131 @@
+        e.execute(new Q(latch));
+        for (int i = 0; i < info.getServices().getMoneypenny(); i++) {
+            e.execute(new Moneypenny(i, latch));
+        }
+        for (int i = 0; i < info.getServices().getM(); i++) {
+            e.execute(new M(i, latch));
+        }
+        for (int i = 0; i < info.getServices().getIntelligence().size(); i++) {
+            e.execute(new Intelligence(i, latch, info.getServices().getIntelligence().get(i).getMissions()));
+        }
+        TimeService timeService = new TimeService(info.getServices().getTime(), latch);
+        e.execute(timeService);
+        while (!timeService.isToTerminate()) {
+            try {
+                timeService.wait();
+            } catch (InterruptedException ignored) {
+            }
+        }
+        e.shutdown();
+        try {
+            // time is not really needed because all threads will die gracefully, and so random time was chosen
+            e.awaitTermination(10, TimeUnit.SECONDS);
+        } catch (InterruptedException ignored) {}
+        try {
+            Inventory.getInstance().printToFile(args[INVENTORY_FILE]);
+            Diary.getInstance().printToFile(args[DIARY_FILE]);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
     }
+
+//    /**
+//     * a method that create all necessary M instances
+//     *
+//     * @param numOfMs num of M instances
+//     *
+//     * @return a list of all the created Ms
+//     */
+//    private static List<M> createMs(int numOfMs) {
+//        List<M> Ms = new ArrayList<>();
+//        for (int i = 0; i < numOfMs; i++) {
+//            Ms.add(new M(i));
+//        }
+//        return Ms;
+//    }
+//
+//    /**
+//     * a method that create all necessary Moneypenny instances
+//     *
+//     * @param numOfMoneypennys num of Moneypenny instances
+//     *
+//     * @return a list of all the created Moneypennys
+//     */
+//    private static List<Moneypenny> createMoneypennys(int numOfMoneypennys) {
+//        List<Moneypenny> Moneypennys = new ArrayList<>();
+//        for (int i = 0; i < numOfMoneypennys; i++) {
+//            Moneypennys.add(new Moneypenny(i));
+//        }
+//        return Moneypennys;
+//    }
+//
+//    /**
+//     * a method that create all necessary Intelligence instances
+//     *
+//     * @param intelligences an array of arrays - the outer array represents intelligences, and the inner array holds the
+//     *                      intelligence's missions.
+//     *
+//     * @return a list of all the created Intelligences
+//     */
+//    private static List<Intelligence> createIntelligences(List<List<MissionInfo>> intelligences) {
+//        List<Intelligence> intelligenceObjects = new ArrayList<>();
+//        for (int i = 0; i < intelligences.size(); i++) {
+//            intelligenceObjects.add(new Intelligence(i, intelligences.get(i)));
+//        }
+//        return intelligenceObjects;
+//    }
```

Comment:
Should these be commented out or just plain removed from the code?

Reply:
I wanted you to see it to decide whether we want this code in the main method, or use the help methods

Reply:
I added some comments on other ways to shorten the main, I think they should be incorporated with the auxiliary functions you added